pkg/download/downloader.go

```diff
@@ -85,20 +88,89 @@ func extractZIP(targetDir string, read io.ReaderAt, size int64) error {
 	return nil
 }
 
+// extractTARGZ extracts a gzipped tar file into the target directory.
+func extractTARGZ(targetDir string, in io.Reader) error {
+	glog.V(4).Infof("tar: extracting to %q", targetDir)
+
+	gzr, err := gzip.NewReader(in)
+	if err != nil {
+		return fmt.Errorf("failed to create gzip reader: %+v", err)
+	}
+	defer gzr.Close()
+
+	tr := tar.NewReader(gzr)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("tar extraction error: %+v", err)
+		}
+		glog.V(4).Infof("tar: processing %q (type=%d, mode=%s)", hdr.Name, hdr.Typeflag, os.FileMode(hdr.Mode))
+		// see https://golang.org/cl/78355 for handling pax_global_header
+		if hdr.Name == "pax_global_header" {
+			glog.V(4).Infof("tar: skipping pax_global_header file")
+			continue
+		}
+
+		path := filepath.Join(targetDir, filepath.FromSlash(hdr.Name))
+		switch hdr.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(path, os.FileMode(hdr.Mode)); err != nil {
+				return fmt.Errorf("failed to create directory from tar: %+v", err)
```

Comment:
, err: %+v .. also for the rest of the fmt calls

Reply:
I don't think the `err:` in  `failed to do x, err: %+v` is necessary.